janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -498,58 +512,112 @@ private static Query numericQuery(String key, Cmp relation, Number value) {
         }
     }
 
-    private SearchParams convertQuery(Condition<?> condition, KeyInformation.StoreRetriever information) {
-        SearchParams params = new SearchParams();
+    // adapted from SolrIndex
+    private List<String> customTokenize(Analyzer analyzer, String fieldName, String value) {
+        final List<String> terms = new ArrayList<>();
+        try (CachingTokenFilter stream = new CachingTokenFilter(analyzer.tokenStream(fieldName, value))) {
+            final TermToBytesRefAttribute termAtt = stream.getAttribute(TermToBytesRefAttribute.class);
+            stream.reset();
+            while (stream.incrementToken()) {
+                terms.add(termAtt.getBytesRef().utf8ToString());
+            }
+            return terms;
+        } catch (IOException e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
+        }
+    }
+
+    private void tokenize(SearchParams params, final Mapping mapping, final LuceneCustomAnalyzer delegatingAnalyzer, String value, String key, JanusGraphPredicate janusgraphPredicate) {
+        final Analyzer analyzer = delegatingAnalyzer.getWrappedAnalyzer(key);
+        final List<String>    terms = customTokenize(analyzer, key, value);
+        if (terms.isEmpty()) {
+            // This might happen with very short terms
+            if (janusgraphPredicate == Text.CONTAINS_PREFIX ) {
```

Comment:
Since this implementation is new (right?) we do need a test to cover it as it's not currently covered by existing tests.

Reply:
Alternatively if this isn't necessary for the custom analyzer feature you're adding here you can move this implementation off to a future PR.

Reply:
I chose to remove it for this pr because the removed code was adressing an ambiguous corner case for example when you are looking for something like "To" in "Tom and Jerry" with Text_Contain_Prefix...
As "to" is a stopword for the English analyzer, "To" gets tokenized to nothing though it seems legitimate to search for things starting with to.

But tokenizing is a tricky business (with normalizing, steming and so on), so I am not even sure that "Tom" is tokenized as "tom" (I think the English analyzer tokenizes it to "tam"...I should check as I'm not 100% sure)  and so, looking for tokens begining with the provided string is probably just a band-aid on a more fondamental issue (tokens are lost sometimes).

And so, as I'm not 100% sure this code is necessary, a decent solution to the problem or working as it should, I removed it.

Thanks for pointing at it. :)